src/psyclone/psyir/nodes/array_reference.py

```diff
@@ -99,6 +99,21 @@ def create(symbol, indices):
             array.addchild(child)
         return array
 
+    def __eq__(self, other):
+        '''
+        Checks whether two nodes are equal. Two ArrayReference nodes are equal
+        if they have the same children (inherited check from parent) and the
+        same symbol.
+
+        :param object other: the object to check equality to.
+
+        :returns: whether other is equal to self.
+        :rtype: bool
+        '''
+        is_eq = super(ArrayReference, self).__eq__(other)
+        is_eq = is_eq and self.symbol == other.symbol
+        return is_eq
+
```

Comment:
No, the super is ArrayMixin I think (which doesn't check this). I think this might be uncovered by tests though so I'll try to remove it once I've sorted the tests if so.